app/lib/bing_api.rb

```diff
@@ -0,0 +1,54 @@
+require 'net/https'
+require 'uri'
+require 'json'
+require 'pry'
+
+class BingApiV7
+  def initialize
+    @accessKey = Rails.application.secrets.bing_api_v7
+  end
+
+  def search(company)
+
+    uri  = "https://api.cognitive.microsoft.com"
+    path = "/bing/v7.0/search"
+
+    if @accessKey.length != 32 then
+      puts "Invalid Bing Search API subscription key!"
+      puts "Please paste yours into the source code."
+      abort
+    end
+
+    uri = URI(uri + path + "?q=" + URI.escape(company))
+
+    puts "Searching the Web for: " + company
+
+    request = Net::HTTP::Get.new(uri)
+    request['Ocp-Apim-Subscription-Key'] = @accessKey
+
+    response = Net::HTTP.start(uri.host, uri.port, :use_ssl => uri.scheme == 'https') do |http|
+      http.request(request)
+    end
+
+    puts "\nRelevant Headers:\n\n"
+    response.each_header do |key, value|
+      # header names are coerced to lowercase
+      if key.start_with?("bingapis-") or key.start_with?("x-msedge-") then
+        puts key + ": " + value
+      end
+    end
+
+    puts "\nJSON Response:\n\n"
+    JSON(response.body)
+  end
+
+  def bing_pages_to_model(company_id)
+    company = Company.find_by(id: company_id)
+    pages = search(company.domain)["webPages"]["value"]
```

Comment:
what if `search(company.domain)` returns nil or empty hash? - CRASH

Reply:
just tested, it works.

Reply:
tested as happy way)

Reply:
what works? What if BING server doesn't work? What if no results? What if no results in webpages?

Also, where is spec file? Use allow+returns to mock call to bing and test call when search returns nil or empty hash.

PS: in fullcontact PR it's already some sample how to mock external calls